### PR TITLE
Add explanation for get_storage in the API reference

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Motivation

Add explanation for get_storage in the API reference.

### CODE REVIEW

File: docs/source/reference/storages.rst [hunk 1]
```diff
@@ -3,11 +3,12 @@
 optuna.storages
 ===============
 
-The :mod:`~optuna.storages` module defines a :class:`~optuna.storages.BaseStorage` class which abstracts a backend database and provides library-internal interfaces to read/write histories of studies and trials. Library users who wish to use storage solutions other than the default in-memory storage should use one of the child classes of :class:`~optuna.storages.BaseStorage` documented below.
+The :mod:`~optuna.storages` module defines a :class:`~optuna.storages.BaseStorage` class which abstracts a backend database and provides library-internal interfaces to read/write histories of studies and trials. Library users who wish to use storage solutions other than the default in-memory storage should instantiate one of the child classes of :class:`~optuna.storages.BaseStorage` documented below by `~optuna.storages.get_storage`.
```
Comment (himkt) [inline docs/source/reference/storages.rst:6]:
```suggestion
The :mod:`~optuna.storages` module defines a :class:`~optuna.storages.BaseStorage` class which abstracts a backend database and provides library-internal interfaces to read/write histories of studies and trials. Library users who wish to use storage solutions other than the default in-memory storage should instantiate one of the child classes of :class:`~optuna.storages.BaseStorage` documented below. You can instantiate a storage by `~optuna.storages.get_storage`.
```

[IMO] It's not a strong opinion but how about dividing a sentence into two passages?

**Review by himkt — COMMENTED** [review]
Thank you @keisuke-umezawa for improving the documentation!
How about adding docstring to `get_storage`?

Example:

```diff
diff --git a/optuna/storages/__init__.py b/optuna/storages/__init__.py
index 25098f80..ad584e62 100644
--- a/optuna/storages/__init__.py
+++ b/optuna/storages/__init__.py
@@ -17,7 +17,15 @@ __all__ = [
 
 
 def get_storage(storage: Union[None, str, BaseStorage]) -> BaseStorage:
+    """Create new storage given storage configuration.
 
+    Args:
+        storage: storage configuration. If `None`, it returns an in-memory storage.
+        If storage is string, it returns a storage according to given storage name.
+        If storage is :class:`~optuna.storages._base.BaseStorage`, it returns
+        a cached storage that wraps an input storage.
+
+    """
     if storage is None:
         return InMemoryStorage()
     if isinstance(storage, str):
```

**Comment by HideakiImamura** [comment]
@Crissman Could you review this PR if you have time?

**Comment by keisuke-umezawa** [comment]
@himkt @Crissman 
Thank you for reviewing, and sorry because this method is designed to be "internal" interface. Most changes will be reverted.

**Review by Crissman — APPROVED** [review]
Oops. Ok, approving.

**Review by himkt — APPROVED** [review]

**Pull request merged**